libs/wingc/src/lib.rs

```diff
@@ -281,10 +282,15 @@ pub fn compile(
 	let default_out_dir = PathBuf::from(format!("{}.out", file_name));
 	let out_dir = out_dir.unwrap_or(default_out_dir.as_ref());
 
+	// Build our AST
+	let (scope, parse_diagnostics) = parse(&source_path);
+
+	// Transform all inflight closures defined in preflight into single-method resources
```

Comment:
This is our "desugaring" phase.

I wouldn't over design this right now but in the future there could be more that we would want to put.

Maybe just add a header comment that says this is where desugaring happens